spec/index.bs

```diff
@@ -2959,7 +2967,17 @@ The {{GPUBufferUsage}} flags determine how a {{GPUBuffer}} may be used after its
 
                     1. Set |b|.{{GPUBuffer/[[deviceTimelineState]]}} to "[=buffer device timeline state/available=]".
 
-                1. Set each byte of |b|'s allocation to zero.
+                1. If |mapOutOfMemory| is `true`,
```

Comment:
This is a little confusing, since in most cases where errors are generated we also stop execution, but here it looks like you intend for the algorithm to continue executing even if the mapping buffer failed with an OOM? My reading on that is that a single call could, in extraordinary circumstances, generate two errors. Not sure if that causes any problems, but it feels weird.

Maybe it's just an order of operations thing? Try the buffer allocation and error+stop if it fails, since that's arguably the more important issue. Then if that succeeds, raise any necessary errors about the mapping buffer failing to allocate.

Reply:
oh, I meant to have a return here. I think this should be first because invalid buffers shouldn't have device allocations associated with them.